[MPS][BE] Turn `exec_unary_kernel` as MetalShaderLibrary method

malfet opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #147299
* #147297
* #147296

And delete duplicate implementations from SpecialOps and UnaryKernel.
Change input and output arguments order for SpecialOps kernels to match those of UnaryOps

Fixes https://github.com/pytorch/pytorch/issues/146770

malfet committed:
[MPS][BE] Turn `exec_unary_kernel` as MetalShaderLibrary class

And delete duplicate implementation sfrom SpecialOps and UnaryKernel

Fixes https://github.com/pytorch/pytorch/issues/146770

[ghstack-poisoned]

dcci reviewed (approved):
Looks like i0 and i1 are currently failing, but the general idea looks great to me. Thank you for fixing this.

malfet commented:
@pytorchbot merge -f *Lint + MPS are green*

malfet commented:
> Looks like i0 and i1 are currently failing, but the general idea looks great to me. Thank you for fixing this.

Argument order is inverted for those I suspect...

malfet committed:
Update on "[MPS][BE] Turn `exec_unary_kernel` as MetalShaderLibrary method"


And delete duplicate implementations from SpecialOps and UnaryKernel

Fixes https://github.com/pytorch/pytorch/issues/146770

[ghstack-poisoned]

malfet commented:
@pytorchbot merge -f "Lint + MPS are green"
